src/verify.rs

```diff
@@ -76,3 +87,27 @@ fn get_ec_curve_sha(
         None
     }
 }
+
+/// Find the verification algorithm for the given RSA-PSS parameters
+///
+/// Not all algorithms are supported, we are limited to what `ring` supports.
+/// Notably, the SHA-1 hash algorithm is not supported.
+fn get_rsa_pss_verification_algo(
+    params: &Option<Any>,
+) -> Option<&'static dyn ring::signature::VerificationAlgorithm> {
+    use ring::signature;
+
+    let params = params.as_ref()?;
+    let params = RsaSsaPssParams::try_from(params).ok()?;
+    let hash_algo = params.hash_algorithm_oid();
```

Comment:
optional nit: I think some of the intermediate bindings might not add much value here:

```suggestion
    let hash_algo = RsaSsaPssParams::try_from(params.as_ref()?)
        .ok()?
        .hash_algorithm_oid();
```

Reply:
Functionally speaking, they are not required, but I think they makes debugging easier, and sometimes also easier fights with the compiler types engine :)